### PR TITLE
add new packages `edge` and `kv`

### PR DESCRIPTION
**Pull request by aead**
This commit introduces two new packages:
 - `kv`
 - `edge`

The `edge` package implements decoding KES
edge server configuration into Go types.

The `kv` package implements generic abstractions
for key stores used by the KES edge server.

Further, the `edge` package contains additional tests for decoding KES configuration.

### CODE REVIEW

File: kv/error.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+// Unwrap returns the Unreachable's underlying error,
+// if any.
+func (e *Unreachable) Unwrap() error { return e.Err }
+
+// Timeout reports whether the Unreachable error
+// is caused by a network timeout.
+func (e *Unreachable) Timeout() bool {
+	if err, ok := e.Err.(net.Error); ok {
+		return err.Timeout()
```
Comment (krisis) [inline kv/error.go:45]:
```suggestion
       var nErr net.Error
        if errors.As(e.Err, &nErr) {
            return nErr.Timeout()
```

**Review by krisis — DISMISSED** [review]
Overall, these changes look good to me, except a minor comment regarding dealing with errors.

**Review by Praveenrajmani — APPROVED** [review]
Can you fix the lint errors? @aead

**Comment by aead** [comment]
Done @Praveenrajmani

**Pull request merged**